### PR TITLE
Fix some test failures

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: lxd/task/group.go [hunk 1]
```diff
@@ -37,16 +37,31 @@ func (g *Group) Start() {
 	ctx := context.Background()
 	ctx, g.cancel = context.WithCancel(ctx)
 	g.wg.Add(len(g.tasks))
-	g.running = make(map[int]bool)
+
+	g.mu.Lock()
+	if g.running == nil {
+		g.running = make(map[int]bool)
+	}
+	g.mu.Unlock()
```
Comment (freeekanayaka) [inline lxd/task/group.go]: Why is this needed? AFAIU there shouldn't be anything else accessing g.running concurrently.

File: lxd/task/group.go [hunk 2]
```diff
@@ -37,16 +37,31 @@ func (g *Group) Start() {
...
+
 	for i := range g.tasks {
-		task := g.tasks[i] // Local variable for the closure below.
+		g.mu.Lock()
+		if g.running[i] {
+			g.mu.Unlock()
+			continue
```
Comment (freeekanayaka) [inline lxd/task/group.go]: I'm confused by this as well.

File: lxd/task/group.go (hunk 1 above)
Comment (freeekanayaka) [reply lxd/task/group.go]: I mean, I believe the way these APIs are supposed to be use is to call a bunch of Group.Add() and then Group.Start(), without any concurrency.

Comment (stgraber) [reply lxd/task/group.go]: Yeah, it's a bit confusing but we've been getting Jenkins failures that show concurrent access to this map, so figured I'd make it safe to call Start() twice and even twice at the same time in theory.

I suspect it's the cluster set of tasks that's hitting this where we're somehow racing between them starting up on startup and LXD starting them after join? I don't know how exactly that's possible unless we manage to process a request for cluster join before we'd normally start cluster tasks, causing them to get started twice?

Comment (freeekanayaka) [reply lxd/task/group.go]: I would add a comment explaining all this. At some point we might look at this and understand better what's going on.

Comment (stgraber) [reply lxd/task/group.go]: Well, I can see how it might actually be useful to be able to add more tasks to an existing group and then call Start() to get those started.

That's effectively what I first attempted to do when splitting the cluster tasks but instead I ended up making two distinct sets of tasks. This change here should make it safe to call Start() twice and also call Add in between to add more tasks which seems potentially useful.
